assets/src/components/detours/diversionPage.tsx

```diff
@@ -365,6 +373,25 @@ export const DiversionPage = ({
               ) : null}
             </ActivateDetour.Modal>
           ) : null}
+
+          {snapshot.matches({
+            "Detour Drawing": {
+              "Share Detour": "Deleting",
+            },
+          }) ? (
+            <DeleteDetourModal
+              onDelete={() =>
+                send({ type: "detour.delete.delete-modal.delete-draft" })
+              }
+              onCancel={() =>
+                send({ type: "detour.delete.delete-modal.cancel" })
+              }
+              routeName={routeName || "??"}
+              routeDescription={routeDescription || "??"}
+              routeOrigin={routeOrigin || "??"}
+              routeDirection={routeDirection || "??"}
+            />
```

Comment:
nit(non-blocking): what do you think of doing this via composistion instead of taking the various props?

```
          affectedRoute={
            <AffectedRoute
              routeName={routeName ?? "??"}
              routeDescription={routeDescription ?? "??"}
              routeOrigin={routeOrigin ?? "??"}
              routeDirection={routeDirection ?? "??"}
            />
          }
```